### DISCUSSION TITLE
Error running Generalized HMM model

### DISCUSSION THREAD

**Discussion by Zehidul-EcoMove** [bmcclintock/momentuHMM#148] [Q&A]
Hi,
I am running 3-state HMM models for tigers. I had successfully run the conventional HMM models 
where temperature and time of day covariates were found to be the best fit model. However, when 
I run a generalized HMM, where I have used the habitat (binary categorical value) effect on the 
state-dependent distribution parameters along with temperature and time of day on transition probability, 
I have the following error. Can you please identify the issue???

Generalized HMM model

nbStates <- 3 
dist <- list(step = "gamma", angle = "vm") 
estAngleMean <- list(angle = TRUE) 

DM <- list(step = list(mean = ~ habitat, sd = ~ 1), ### covariate effects
           angle = list(mean = ~ 1, concentration = ~1))

checkPar0(mHMM, nbStates = nbStates, 
          dist = dist, 
          DM = DM, 
          mixtures = 3,
          estAngleMean = estAngleMean)

getPar <- list(step =  c(0.014, 0.105, 0.785, 0.011 , 0.085, 0.565),#gamma
               angle = c(3.14, 0.105, 0.031, 0.065, 0.354, 1.125))#vm

(getPar <- getParDM(data = mHMM,
                    nbStates = nbStates,
                    dist = dist,
                    Par = getPar,
                    estAngleMean = estAngleMean,
                    DM = DM))

Par0 <- list(step = getPar$step, 
             angle = getPar$angle)

formula <- ~ temp + cosinor(hour, period = 24)

generalized_hmm1 <- fitHMM(mHMM,
                   DM=DM,
                   dist=dist,
                   nbStates=nbStates,
                   Par0=Par0,
                   formula=formula,
                   mixtures = 3,
                   estAngleMean=estAngleMean,
                   covNames = "temp", "hour")

Error in fitHMM.momentuHMMData(mHMM, DM = DM, dist = dist, nbStates = nbStates,  : 
  beta0 must be a list with elements named 'beta' and/or 'pi' when mixtures>1
In addition: Warning message:
In fitHMM.momentuHMMData(mHMM, DM = DM, dist = dist, nbStates = nbStates, 
    Par0 = Par0, formula = formula, mixtures = 3, estAngleMean = estAngleMean, 
    covNames = "temp", "hour") :
 extra argument ‘covNames’ will be disregarded

**Comment by bmcclintock** [comment]
I don't see anything that should cause an error in the code you provided, but I can't diagnose the issue without a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example#:~:text=In%20computing%2C%20a%20minimal%20reproducible,to%20be%20demonstrated%20and%20reproduced.). Can you provide one? If so, please also provide the session information:
```
sessionInfo()
```
